src/WLGDPSTime.cc

```diff
@@ -0,0 +1,77 @@
+// WLGDPSTime
+#include "WLGDPSTime.hh"
+#include "G4UnitsTable.hh"
+
+////////////////////////////////////////////////////////////////////////////////
+// Description:
+//   This is a primitive scorer class for scoring global track time
+//
+///////////////////////////////////////////////////////////////////////////////
+
+WLGDPSTime::WLGDPSTime(G4String name, G4int depth)
+: G4VPrimitiveScorer(name, depth)
+, HCID(-1)
+, EvtMap(0)
+{
+    SetUnit("ns");
+}
+
+WLGDPSTime::WLGDPSTime(G4String name, const G4String& unit, G4int depth)
+: G4VPrimitiveScorer(name, depth)
+, HCID(-1)
+, EvtMap(0)
+{
+    SetUnit(unit);
+}
+
+WLGDPSTime::~WLGDPSTime() { ; }
+
+G4bool WLGDPSTime::ProcessHits(G4Step* aStep, G4TouchableHistory*)
+{
+    if(aStep->GetPostStepPoint()->GetStepStatus() == fGeomBoundary)
+        return FALSE;  // all collisions
```

Comment:
```suggestion
    // better comment here - what does "all collisions" mean?
    if(aStep->GetPostStepPoint()->GetStepStatus() == fGeomBoundary)
    {
        return FALSE;
    }
```

Reply:
comes straight from the collision primitive scorer, it is the geometry boundary status. Anything not a step to a boundary is a collision.